app/rankers/algorithms.py

```diff
@@ -242,6 +243,19 @@ def user_impression_filter(recs: RecommendationListType, user_impressed_list: Li
     return filtered
 
 
+def rank_by_preferred_topics(
+        recs: CorpusRecommendationListType,
+        preferred_topics: List[TopicModel],
+) -> CorpusRecommendationListType:
+    """
+    Sorts recommendations in-order such that recs with a preferred topic rank above recs with a non-preferred topic.
+    :param recs:
+    :param preferred_topics: List of topics that the user has expressed a preference for.
+    :return: ordered list of recommendations with preferred topics ranking higher.
+    """
+    preferred_corpus_topic_ids = {topic.corpus_topic_id for topic in preferred_topics}
```

Comment:
Nit: This is probably worth a unit test if for no more reason than giving a very specific example.

Reply:
Good call-out. I've added a ticket to track this under 'cleanup' just so I don't forget. Should only take a minute: https://getpocket.atlassian.net/browse/HS-64